### PR TITLE
Syaam/19 settings schema

### PR DESCRIPTION
**Pull request by syaamkhandaker**
## PR Title

Issue Number(s): #19 

What does this PR change and why?

This PR adds the settings schema, api layers, and DAO.

### Checklist

- [x] Requirements have been implemented
- [x] Acceptance criteria is met
- [ ] Database schema docs have been updated or are not necessary
- [x] Code follows design and style guidelines
- [x] Commits follow guidelines (concise, squashed, etc)
- [ ] Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### Critical Changes

- Database change/migration to run
- Environment config change
- Breaking API change

### Related PRs

- #19 
### Testing

Enumerate steps to test the functionality of your ticket. This should include edge cases and testing of error handling, if applicable.

Postman

### CODE REVIEW

**Review by michaelhyi — CHANGES_REQUESTED** [review]
Great work Syaam! This works well. Left some comments!

File: src/db/actions/settings.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import { InternalServerError } from "@/types/exceptions";
+import { ObjectId } from "mongodb";
+import client from "../dbClient";
+import {
+  UpdateSettingsObject,
+  UpdateSettingsPinObject,
+} from "@/types/settings";
+import { Settings } from "../models";
+
+export async function createNewSettings(newSettings: Settings) {
```
[the remark this answers is not in the document]
Comment (syaamkhandaker) [reply src/db/actions/settings.ts:10]: this has to be createnewsettings because the other method name is createSettings, so there's a conflict. I feel like the service method should be create settings too

File: src/services/settings.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  const settings = await getSettingsByUserId(new ObjectId(userId));
+
+  if (!settings) {
+    throw new DoesNotExistError("Settings does not exist for this user");
+  }
+
+  return settings as Settings;
+}
+
+export async function updateSettings({
```
[the remark this answers is not in the document]
Comment (syaamkhandaker) [reply src/services/settings.ts:81]: but all of these attributes are booleans

File: src/services/settings.ts
[the remark this answers is not in the document]
Comment (syaamkhandaker) [reply src/services/settings.ts:1]: What do you mean by dealign with userid?

**Review by michaelhyi — APPROVED** [review]
Let's go!
